File: src/game/systems/snare.lua
local Aspect = require 'ecs.aspect'
local System = require 'ecs.system'
local Snare = require 'game.components.snare'

local aspect = Aspect.new({Snare})
local SnareSystem = System:new('snare', aspect)

function SnareSystem:beginContact(a, b, contact)
end

return SnareSystem

File: client2/src/game/systems/Snare.ts
import { System } from '../../ecs/System'
import { SystemFlag } from '../flags'
import { NeverAspect } from '../../ecs/Aspect'
import { check, hasComponent } from '../utils/collision'
import { Player } from '../components/Player'
import { Snare } from '../components/Snare'

export class SnareSystem extends System {
  static _id = 'Snare'
  _id = SnareSystem._id

  static _flag = SystemFlag.Snare
  _flag = SnareSystem._flag

  static _aspect = new NeverAspect()
  _aspect = SnareSystem._aspect

  beginContact = (a: Fixture, b: Fixture, contact: Contact) => {
    const result = check(a, b, [hasComponent(Player), hasComponent(Snare)])

    if (!result) {
      return
    }
  }
}
